### PR TITLE
Support direct annotation argument (--train/val-ann-files)

### PR DESCRIPTION
**Pull request by sungmanc**
### Summary

* Support direct annotation input for CoCo
For now, we can use `--train/val-ann-files` argument
* Support unlabeled file list argument 
* Add unit-tests for both functionalities
* Unifying the `data_root` + `data_roots` --> `data_roots`
* Edit the CHANGELOG
* Edit the documents
image

* Add unlabeled example assets
* Add subset annotation example assets

### How to test
[To check the direct annotation functionality (Dataset Adapter)]
[To check the unlabeled file functionality]
* tox -e pre-merge-all-py310 -- tests/unit/core/data/adapter/test_init.py

[To check the direct annotation functionality (Datumaro function itself)]
* tox -e pre-merge-all-py310 -- tests/unit/core/data/manager/test_dataset_manager.py

### Checklist

- [X] I submit my changes into the `develop` branch
- [X] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md)
- [X] I have updated the [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) accordingly
- [X] I have added tests to cover my changes
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [X] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [X] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by yunchu** [comment]
please add label for this PR.

File: otx/cli/manager/config_manager.py [hunk 1]
```diff
@@ -257,11 +259,15 @@ def _get_task_type_from_data_format(self, data_format: str) -> str:
                 return task_key
         raise ConfigValueError(f"Can't find proper task. we are not support {data_format} format, yet.")
 
-    def auto_split_data(self, data_roots: str, task: str):
+    def auto_split_data(self, data_roots: str, task: str, train_ann_file: Optional[str] = None):
         """Automatically Split train data --> train/val dataset."""
         self.data_format = self.dataset_manager.get_data_format(data_roots)
         dataset = self.dataset_manager.import_dataset(data_root=data_roots, data_format=self.data_format)
         train_dataset = self.dataset_manager.get_train_dataset(dataset)
```
Comment (harimkang) [inline otx/cli/manager/config_manager.py:266]: Maybe we could move line 266-270 to the line 280 else statement? It looks to me like `train_dataset` is not used in any other case.

Comment (harimkang) [inline otx/cli/manager/config_manager.py:266]: Maybe we could move line 266-270 to the line 280 else statement? It looks to me like train_dataset is not used in any other case.

Comment (harimkang) [reply otx/cli/manager/config_manager.py:266]: Oh, my bad, it's used on line 277.

File: otx/cli/manager/config_manager.py [hunk 2]
```diff
@@ -257,11 +259,15 @@ def _get_task_type_from_data_format(self, data_format: str) -> str:
                 return task_key
         raise ConfigValueError(f"Can't find proper task. we are not support {data_format} format, yet.")
 
-    def auto_split_data(self, data_roots: str, task: str):
+    def auto_split_data(self, data_roots: str, task: str, train_ann_file: Optional[str] = None):
```
Comment (harimkang) [inline otx/cli/manager/config_manager.py]: Couldn't we use `ann_file` instead of `train_ann_file`? (This is simply because the first parameter uses data_roots.)

**Review by harimkang — DISMISSED** [review]
LGTM!

File: otx/core/data/adapter/base_dataset_adapter.py [hunk 3]
```diff
@@ -125,6 +146,8 @@ def _import_dataset(
             self.data_type = self._select_data_type(self.data_type_candidates)
 
             train_dataset = DatumDataset.import_from(train_data_roots, format=self.data_type)
+            if train_ann_files is not None:
+                train_dataset = DatumDataset.import_from(train_ann_files, format=self.data_type, subset="train")
```
Comment (cih9088) [inline otx/core/data/adapter/base_dataset_adapter.py]:
```suggestion
            dataset_kwargs = {"path": train_data_roots, "format": self.data_type}
            if train_ann_files is not None:
                if self.data_type not in ("coco"):
                    raise NotImplementedError(f"Specifying '--train-ann-files' is not supported for data type '{self.data_type}'")
                dataset_kwargs["path"] = train_ann_files
                dataset_kwargs["subset"] = "train"
            train_dataset = DatumDataset.import_from(**dataset_kwargs)
```

File: otx/core/data/adapter/base_dataset_adapter.py [hunk 4]
```diff
@@ -135,6 +158,8 @@ def _import_dataset(
                 val_data_candidates = self._detect_dataset_format(path=val_data_roots)
                 val_data_type = self._select_data_type(val_data_candidates)
                 val_dataset = DatumDataset.import_from(val_data_roots, format=val_data_type)
+                if val_ann_files is not None:
+                    val_dataset = DatumDataset.import_from(val_ann_files, format=self.data_type, subset="val")
```
Comment (cih9088) [inline otx/core/data/adapter/base_dataset_adapter.py]:
```suggestion
                dataset_kwargs = {"path": val_data_roots, "format": val_data_type}
                if val_ann_files is not None:
                    if val_data_type not in ("coco"):
                        raise NotImplementedError(f"Specifying '--val-ann-files' is not supported for data type '{val_data_type}'")
                    dataset_kwargs["path"] = val_ann_files
                    dataset_kwargs["subset"] = "val"
                val_dataset = DatumDataset.import_from(**dataset_kwargs)
```

File: otx/core/data/adapter/base_dataset_adapter.py [hunk 5]
```diff
@@ -143,12 +168,15 @@ def _import_dataset(
             self.data_type_candidates = self._detect_dataset_format(path=test_data_roots)
             self.data_type = self._select_data_type(self.data_type_candidates)
             test_dataset = DatumDataset.import_from(test_data_roots, format=self.data_type)
+            if test_ann_files is not None:
+                test_dataset = DatumDataset.import_from(test_ann_files, format=self.data_type, subset="test")
```
Comment (cih9088) [inline otx/core/data/adapter/base_dataset_adapter.py]:
```suggestion
            dataset_kwargs = {"path": test_data_roots, "format": self.data_type}
            if test_ann_files is not None:
                if self.data_type not in ("coco"):
                    raise NotImplementedError(f"Specifying '--test-ann-files' is not supported for data type '{self.data_type}'")
                dataset_kwargs["path"] = test_ann_files
                dataset_kwargs["subset"] = "test"
            test_dataset = DatumDataset.import_from(**dataset_kwargs)
```

**Review by cih9088 — COMMENTED** [review]
I noticed that `--{subset}--data-roots` and `--{subset}-ann-files` arguments have descriptions saying one can pass comma separated list of paths but it certainly does not support it.

It would be better to rename the plural to singular and modify the description or implement the comma separated list of paths feature.

File: otx/cli/manager/config_manager.py (hunk 2 above)
Comment (sungmanc) [reply otx/cli/manager/config_manager.py]: Done. [fd4b79d](https://github.com/openvinotoolkit/training_extensions/pull/1921/commits/fd4b79d485996d8cdc6c43a902f1ddf25a21a227)

**Comment by sungmanc** [comment]
> I noticed that `--{subset}--data-roots` and `--{subset}-ann-files` arguments have descriptions saying one can pass comma separated list of paths but it certainly does not support it.
> 
> It would be better to rename the plural to singular and modify the description or implement the comma separated list of paths feature.

Agree, then could I change all variable and description at next PR?

**Review by cih9088 — DISMISSED** [review]

**Review by cih9088 — APPROVED** [review]

**Review by harimkang — APPROVED** [review]
